src/servers/buttongestureadapter.ts

```diff
@@ -0,0 +1,128 @@
+import {
+    ButtonEvent,
+    ButtonGestureEvent,
+    CONNECT,
+    DEVICE_CHANGE,
+    DEVICE_CONNECT,
+    EVENT,
+    ROLE_BOUND,
+    SELF_ANNOUNCE,
+    SRV_BUTTON,
+    SRV_BUTTON_GESTURE,
+} from "../jdom/constants"
+import SensorServer from "./sensorserver"
+import { assert, JDDevice, JDEvent, JDService } from "../jdom/jacdac-jdom"
+import RoleManager from "./rolemanager"
+import JDServiceServer from "../jdom/serviceserver"
+
+export class AdapterServer extends JDServiceServer {
+    private roleManager? : RoleManager = undefined
+    
+    // This is a nasty hack that allows the role manager to be set after the server is instantiated.
+    // The right solution would be to have this listen for role broadcasts for something
+    // but that's not a now feature.
+    // TODO de-jankify
+    public _hack_setRoleManager(roleManager: RoleManager) {
+        assert(this.roleManager === undefined, "resetting role manager")
+
+        this.roleManager = roleManager
+        this.onRoleManager(roleManager)
```

Comment:
Use emit instead of declaring handlers

Reply:
Typically CHANGE event when assigning a new role manager

Reply:
And wrap this up using get/set properties. It's w pattern we use commonly, see. BlockServices.value